src/org/usfirst/frc4904/robot/subsystems/Hopper.java

```diff
@@ -0,0 +1,46 @@
+package org.usfirst.frc4904.robot.subsystems;
+
+
+import org.usfirst.frc4904.standard.commands.Idle;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.command.Subsystem;
+
+public class Hopper extends Subsystem {
+	protected final DoubleSolenoid leftSolenoid;
+	protected final DoubleSolenoid rightSolenoid;
+
+	public static enum HopperState {
+		BALLIO(DoubleSolenoid.Value.kReverse), SHOOTER(DoubleSolenoid.Value.kForward);
+		private final DoubleSolenoid.Value value;
+
+		private HopperState(DoubleSolenoid.Value value) {
+			this.value = value;
+		}
+
+		public DoubleSolenoid.Value getValue() {
```

Comment:
Why a getter instead of access to the value field? Having the field be final (as it is) already prevents mutation.